src/components/dialogs/two-windings-transformer/ratio-tap-changer-pane.js

```diff
@@ -0,0 +1,535 @@
+
+    const handleEditCell = useCallback(
+        (rowData, newVal) => {
+            const parsedVal = parseInt(newVal, 10);
+
+            if (!isNaN(parsedVal) && parsedVal >= 0 && parsedVal <= 100) {
+                let tempRows = ratioTapRows;
+                const column = rowData.dataKey;
+                tempRows[rowData.rowIndex][column] = parsedVal;
+                tempRows[rowData.rowIndex].isEdited = true;
+                handleRatioTapRows(tempRows);
+            }
+        },
+        [handleRatioTapRows, ratioTapRows]
+    );
+
+    const editableCellRender = useCallback(
+        (rowData) => {
+            if (isLineOnEditMode(rowData)) {
+                const index = rowData.columnIndex - 1;
+                const Editor = COLUMNS_DEFINITIONS[index].editor;
+                if (Editor) {
+                    return (
+                        <div className={classes.tableCell}>
+                            <Editor
+                                key={rowData.dataKey + index}
+                                columnDefinition={COLUMNS_DEFINITIONS[index]}
+                                defaultValue={rowData.cellData}
+                                setColumnError={(k) => setColumnInError(k)}
+                                resetColumnError={(k) => resetColumnInError(k)}
+                                setter={(val) => handleEditCell(rowData, val)}
+                            />
+                        </div>
+                    );
+                }
+            }
+            return defaultCellRender(rowData);
+        },
+        [
+            COLUMNS_DEFINITIONS,
+            classes.tableCell,
+            defaultCellRender,
+            handleEditCell,
+            isLineOnEditMode,
+            resetColumnInError,
+            setColumnInError,
+        ]
+    );
+
+    const generateTableColumns = () => {
+        let tableColumns = Object.values(COLUMNS_DEFINITIONS).map((c) => {
+            if (c.editor) {
+                c.cellRenderer = editableCellRender;
+            }
+            return c;
+        });
+
+        tableColumns.unshift({
+            id: 'edit',
+            dataKey: 'edit',
+            label: '',
+            minWidth: 75,
+            locked: true,
+            editColumn: true,
+            cellRenderer: (rowData, rowIndex) =>
+                editCellRender(rowData, rowIndex),
+        });
+        return tableColumns;
+    };
+
+    const getCSVColumns = () => {
+        return [
+            intl.formatMessage({ id: 'Tap' }),
+            intl.formatMessage({ id: 'ImportFileResistance' }),
+            intl.formatMessage({ id: 'ImportFileReactance' }),
+            intl.formatMessage({ id: 'ImportFileConductance' }),
+            intl.formatMessage({ id: 'ImportFileSusceptance' }),
+            intl.formatMessage({ id: 'Ratio' }),
+        ];
+    };
+
+    const parseIntData = (data, defaultValue) => {
+        return isNaN(parseInt(data), 10) ? defaultValue : parseInt(data);
```

Comment:
Uh : `isNan(parseInt(data), 10)`, I guess the first closing parenthesis comes too early.
You could have saved the result of parseInt, to return it after having tested it.
I doubt the compiler can optimize this, especially here ;-)

Reply:
Weird it didn't caused an error.
I've stored parseInt(data, 10) in a const then

Reply:
*Just parseInt(data) since the radix default value is already 10